gitlab-runner/README.md

```diff
@@ -0,0 +1,98 @@
+# gitlab-runner-rs
+
+A crate to help build custom gitlab runner implementations.
+
+## Implementing a custom runner
+
+The overall idea is that this crate handles all interaction with the gitlab
+server and drives the executions while the the runner implementation focus on
+how to handle jobs from gitlab.
+
+As the focus for an implementer of a custom runner is to implement the async
+JobHandler trait, which gets calle during job executation. An absolute minimal
+runner can be implement as such:
+
+```rust,no_run
+use gitlab_runner::{outputln, GitlabLayer, RunnerBuilder, JobHandler, JobResult, Phase};
+use tracing_subscriber::prelude::*;
+use std::path::PathBuf;
+
+#[derive(Debug)]
+struct Run {}
+
+#[async_trait::async_trait]
+impl JobHandler for Run {
+      async fn step(&mut self, script: &[String], phase: Phase) -> JobResult {
+          outputln!("Running script for phase {:?}", phase);
+          for s in script {
+            outputln!("Step: {}", s);
+          }
+          Ok(())
+      }
+}
+
+#[tokio::main]
+async fn main() {
+    let (layer, jobs) = GitlabLayer::new();
+
+     // By default show logs from my_runner and warnings, the gitlab_runner::gitlab::job span which
+     // sets the gitlab.job field is always enabled regardless of the over all log level.
+    let env = std::env::var("RUNNER_LOG") .unwrap_or_else(|_| "my_runner=info,warn".to_string());
+    let envfilter = tracing_subscriber::EnvFilter::builder()
+        .parse(env)
+        .expect("Failed to parse RUNNER_LOG env var: {env}")
```

Comment:
Missing a `format!` call?

```suggestion
        .expect(format!("Failed to parse RUNNER_LOG env var: {env}"))
```

Reply:
It's not performance critical; but `expect(format!` is not great in principle as that will always create a String; so replaced this with `.unwrap_or_else(|| panic!(...))` instead :)